### PR TITLE
feat(contracts-periphery): Make optimist deploy scripts safer

### PR DESCRIPTION
**Pull request by roninjin10**
- add sanity checks in post deployment
- add some typesafety

### CODE REVIEW

**Comment by roninjin10** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #4454** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/4454" target="_blank">Graphite</a> 
    * **PR #4455** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/4455" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/4455?utm_source=stack-comment).

**Comment by tynes** [comment]
This will cause a conflict with https://github.com/ethereum-optimism/optimism/pull/4437

We want to do deterministic deployment, and there is no post hook with the deterministic deployment method, so I recommend moving the checks to the bottom of the file instead of in the post hook

**Comment by roninjin10** [comment]
I'll restack on #4437

File: packages/contracts-periphery/deploy/op-nft/OptimistProxy.ts [hunk 1]
```diff
@@ -86,15 +90,20 @@ const deployFn: DeployFunction = async (hre: HardhatRuntimeEnvironment) => {
     'AttestationStationProxy'
   )
 
-  assertContractVariable(Optimist, 'name', hre.deployConfig.optimistName)
-  assertContractVariable(Optimist, 'symbol', hre.deployConfig.optimistSymbol)
-  assertContractVariable(
+  await assertContractVariable(Optimist, 'admin', deployer)
```
Comment (tynes) [inline packages/contracts-periphery/deploy/op-nft/OptimistProxy.ts:93]: Does this work?

**Comment by tynes** [comment]
We need to merge this before deployment

File: packages/contracts-periphery/deploy/op-nft/OptimistProxy.ts (hunk 1 above)
Comment (roninjin10) [reply packages/contracts-periphery/deploy/op-nft/OptimistProxy.ts:93]: bad rebase
```suggestion
```

File: packages/contracts-periphery/deploy/op-nft/OptimistProxy.ts [hunk 2]
```diff
@@ -86,15 +90,20 @@ const deployFn: DeployFunction = async (hre: HardhatRuntimeEnvironment) => {
...
+  await assertContractVariable(Optimist, 'name', deployConfig.optimistName)
```
[the remark this answers is not in the document]
Comment (tynes) [reply packages/contracts-periphery/deploy/op-nft/OptimistProxy.ts:94]: can also

```
const expectedAdminSlot = hre.ethers.utils.hexZeroPad(deployConfig.admin, 32)
```

**Review by smartcontracts — APPROVED** [review]

**Pull request merged**